solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -328,8 +328,8 @@ impl IbcStorage<'_, '_> {
         }
 
         let denom = coin.denom.to_string();
-        let denom = denom.rsplit_once('/').unwrap_or((&denom, &denom)).1;
-        let escrow = get_escrow_account(port_id, channel_id, &denom);
+        let denom = denom.rsplit_once('/').unwrap_or(("", &denom)).1;
+        let escrow = get_escrow_account(port_id, channel_id, denom);
```

Comment:
I believe this can now just be
```sugestion
        let escrow = get_escrow_account(port_id, channel_id, coin.denom.base_denom.as_str());
```

Reply:
well, the issue still persists because while unescrowing ( when we receive the packet ), we only get the `base_denom`. So we either use `base_denom` or we try to add prefix which i am wondering how since we dont have the source channel id.

Reply:
Right, but I mean at this point this is equivalent. We’re taking only
the part after final slash which is base denom. So we can just take
coin.denom.base_denom.  Though I guess it’s worth adding a TODO:

```rust
        // TODO(#180): Should we use full denom including prefix?
        let escrow = get_escrow_account(port_id, channel_id, coin.denom.base_denom.as_str());
```

Reply:
sure sure, sounds good, will add it